smarts/core/sumo_road_network.py

```diff
@@ -946,28 +955,41 @@ def _snap_external_holes(self, lane_to_poly, snap_threshold=2):
             if lane.getEdge().isSpecial():
                 continue
 
-            incoming = self._graph.getLane(lane_id).getIncoming()
+            incoming = lane.getIncoming()
             if incoming and incoming[0].getEdge().isSpecial():
                 continue
 
-            outgoing = self._graph.getLane(lane_id).getOutgoing()
+            outgoing = lane.getOutgoing()
             if outgoing:
                 outgoing_lane = outgoing[0].getToLane()
                 if outgoing_lane.getEdge().isSpecial():
                     continue
 
             lane_shape = lane_to_poly[lane_id]
+            new_coords = []
             for x, y in lane_shape.exterior.coords:
                 for nl, dist in self.nearest_lanes(
                     Point(x, y),
                     include_junctions=False,
                 ):
-                    if (not nl) or (nl and nl.in_junction):
+                    if not nl or nl.in_junction or nl.lane_id == lane_id:
+                        continue
+                    ll, _ = nl.lane_to_left
+                    rl, _ = nl.lane_to_right
+                    if (not ll or ll.lane_id != lane_id) and (
+                        not rl or rl.lane_id != lane_id
+                    ):
```

Comment:
Is there any way to just use `lane_to_left|lane_to_right` exclusively instead of also calling `nearest_lanes`? I feel as though left/right by itself would be faster to resolve.

I think this also might now be bypassing the case of outgoing and in-going lanes although I suspect they should not be a problem in.